### PR TITLE
Make seed an option

### LINKED ISSUE #7: Generate seed when no seed is provided
Make the param `Option` type

### PR DESCRIPTION
**Pull request by matias-gonz**
Closes #7

### CODE REVIEW

File: src/mega_lehmer.rs [hunk 1]
```diff
@@ -1,12 +1,16 @@
-use crate::felt::FeltTrait;
+use crate::{felt::FeltTrait, seed_generators::time_seed};
```
Comment (RobertEggert) [inline src/mega_lehmer.rs:1]: add multiplier gen (it is in my branch i guess i will change what you did and then pull request it, then we can add it)

File: src/mega_lehmer.rs [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
     last_gen: F,
 }
 
 impl<F: FeltTrait> MegaLehmer<F> {
-    pub fn new(seed: F, multiplier: F) -> MegaLehmer<F> {
+    pub fn new(seed: Option<F>, multiplier: F) -> MegaLehmer<F> {
+        let seed = match seed {
+            Some(seed) => seed,
+            None => time_seed(),
+        };
```
Comment (RobertEggert) [inline src/mega_lehmer.rs:13]: Same as in my comment (what if we have more gens?)

Comment (matias-gonz) [reply src/mega_lehmer.rs:13]: We first need a `SeedGeneratorTrait`. This is the only thing we have for now. We will probably use a generator as a param

File: src/mega_lehmer.rs [hunk 3]
```diff
@@ -28,7 +32,7 @@ mod tests {
 
     #[test]
     fn test_gen() {
-        let mut lehmer = MegaLehmer::new(Felt17::new(1), Felt17::new(2));
+        let mut lehmer = MegaLehmer::new(Some(Felt17::new(1)), Felt17::new(2));
```
Comment (RobertEggert) [inline src/mega_lehmer.rs:35]: both have to be an option as we can either generate or give something as input .. in my branch i already did a generator, so maybe do this also as an option type

**Review by RobertEggert — APPROVED** [review]
Rest looks good:)

File: src/mega_lehmer.rs (hunk 3 above)
Comment (matias-gonz) [reply src/mega_lehmer.rs:35]: The multiplier cant be an option since we dont have a generator yet

**Pull request merged**